telescope/ERQTelescopePID.cxx

```diff
@@ -189,89 +179,103 @@ Double_t CalcElossIntegralVolStep (Double_t T, G4ParticleDefinition* ion,
 }
 
 //--------------------------------------------------------------------------------------------------
-Double_t ERQTelescopePID::CalcEloss(TString station, ERQTelescopeTrack* track, Int_t pdg, Double_t T){
+TVector3 ERQTelescopePID::FindBackPropagationStartPoint(const ERQTelescopeTrack* track) {
+  const TVector3 telescopeVertex = track->GetTelescopeVertex();
+  TVector3 backPropagationStartPoint = telescopeVertex;
+  TVector3 direction = telescopeVertex - track->GetTargetVertex();
+  direction.SetMag(1.);
+  TGeoNode* currentNode = gGeoManager->InitTrack(telescopeVertex.X(), telescopeVertex.Y(), telescopeVertex.Z(),
+                                                 direction.X(), direction.Y(), direction.Z());
+  TGeoNode* lastSensetiveNode = nullptr;
+  TString lastSensetivePath;
+  TVector3 lastSensetivePosition;
+  while(!gGeoManager->IsOutside()) {
+    bool inSensetiveVolume = false;
+    const TString path = gGeoManager->GetPath();
+    if (path.Contains("Sensitive")) {
+      inSensetiveVolume = true;
+      lastSensetiveNode = currentNode;
+      lastSensetivePath = path; 
+    }
+    currentNode = gGeoManager->Step();
```

Comment:
gGeoManager->FindNextBoundaryAndStep() should be used instead gGeoManager->Step(). It's more confident and faster. I had bugs with using the Step() here: the last sensitive volume has was found only in the first appropriate event, in other correct events algorithm couldn't find it for some reason.